tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedInsertManyOperation.cs

```diff
@@ -0,0 +1,127 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using MongoDB.Driver.Tests.Specifications.unified_test_format;
+
+namespace MongoDB.Driver.Tests.UnifiedTestOperations
+{
+    public class UnifiedInsertManyOperation : IUnifiedEntityTestOperation
+    {
+        private readonly IMongoCollection<BsonDocument> _collection;
+        private readonly List<BsonDocument> _documents;
+        private readonly InsertManyOptions _options;
+        private readonly IClientSessionHandle _session;
+
+        public UnifiedInsertManyOperation(
+            IClientSessionHandle session,
+            IMongoCollection<BsonDocument> collection,
+            List<BsonDocument> documents,
+            InsertManyOptions options)
+        {
+            _session = session;
+            _collection = collection;
+            _documents = documents;
+            _options = options;
+        }
+
+        public OperationResult Execute(CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (_session == null)
+                {
+                    _collection.InsertMany(_documents, _options, cancellationToken);
+                }
+                else
+                {
+                    _collection.InsertMany(_session, _documents, _options, cancellationToken);
+                }
+
+                return OperationResult.FromResult(new BsonDocument()); // In .NET InsertMany returns no result
```

Comment:
why not

    return OperationResult.FromResult(null);

like in `UnifiedInsertOneOperation`?

Reply:
This is a workaround because of test file's `expectResult`:
```
"expectResult":
{
  "insertedCount":
  {
    "$$unsetOrMatches": 2
  },
  "insertedIds":
  {
    "$$unsetOrMatches":
    {
    "0": 3,
    "1": 4
    }
  }
}
```
Which means that `UnifiedValueMatcher` will raise an error if the document is `null`: [UnifiedValueMatcher.cs#L86](https://github.com/MikalaiMazurenka/mongo-csharp-driver/blob/ceea35f605b2f428fb5e964982e6f29985d12938/tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedValueMatcher.cs#L86)
This can be fixed if we cover the whole `expectResult` with `$$unsetOrMatches` like this:
```"expectResult":
{
  "$$unsetOrMatches":
  {
    "insertedCount":
    {
      "$$unsetOrMatches": 2
    },
    "insertedIds":
    {
      "$$unsetOrMatches":
      {
        "0": 3,
        "1": 4
      }
    }
  }
}
```
So I've requested a change in the spec.
Done